src/index.spec.ts

```diff
@@ -1,12 +1,227 @@
+    };
+    const testFunction3 = () => {
+      createProduct(product3);
+    };
+    const testFunction4 = () => {
+      createProduct(product4);
+    };
+    const testFunction5 = () => {
+      createProduct(product5);
+    };
+    expect(testFunction2).toThrow(Error);
+    expect(testFunction3).toThrow(Error);
+    expect(testFunction4).toThrow(Error);
+    expect(testFunction5).toThrow(Error);
+  });
+});
+
+describe('createRandomProduct test', () => {
+  it('should return a product with correct email', () => {
+    expect(createRandomProduct('clark@kent.com')).toMatchObject({
+      id: expect.anything(),
+      name: expect.anything(),
+      description: expect.anything(),
+      price: expect.anything(),
+      tags: expect.anything(),
+    });
+  });
+  it('should return Error with a email without create rol', () => {
+    const createWithAdmin = () => {
+      createRandomProduct('bruce@wayne.com');
+    };
+    const createWithUser = () => {
+      createRandomProduct('diana@themyscira.com');
+    };
+    const createWithEmpty = () => {
+      createRandomProduct('');
+    };
+    expect(createWithAdmin).toThrow(Error);
+    expect(createWithUser).toThrow(Error);
+    expect(createWithEmpty).toThrow(Error);
+  });
+});
+
+describe('getStarWarsPlanets function test', () => {
+  const response = {
+    count: 60,
+    next: 'https://swapi.dev/api/planets/',
+    previous: null,
+    results: expect.any(Array),
+  };
+
+  const mockedResponse = {
+    json: () =>
+      Promise.resolve({
+        count: 60,
+        next: 'https://swapi.dev/api/planets/',
+        previous: null,
+        results: [
+          {
+            name: 'Tatooine',
+            rotation_period: '23',
+            orbital_period: '304',
+            diameter: '10465',
+            climate: 'arid',
+            gravity: '1 standard',
+            terrain: 'desert',
+            surface_water: '1',
+            population: '200000',
+            residents: [
+              'https://swapi.dev/api/people/1/',
+              'https://swapi.dev/api/people/2/',
+              'https://swapi.dev/api/people/4/',
+              'https://swapi.dev/api/people/6/',
+              'https://swapi.dev/api/people/7/',
+              'https://swapi.dev/api/people/8/',
+              'https://swapi.dev/api/people/9/',
+              'https://swapi.dev/api/people/11/',
+              'https://swapi.dev/api/people/43/',
+              'https://swapi.dev/api/people/62/',
+            ],
+            films: [
+              'https://swapi.dev/api/films/1/',
+              'https://swapi.dev/api/films/3/',
+              'https://swapi.dev/api/films/4/',
+              'https://swapi.dev/api/films/5/',
+              'https://swapi.dev/api/films/6/',
+            ],
+            created: '2014-12-09T13:50:49.641000Z',
+            edited: '2014-12-20T20:58:18.411000Z',
+            url: 'https://swapi.dev/api/planets/1/',
+          },
+        ],
+      }),
+  };
```

Comment:
This big object could be moved to a new file named `mock.ts` or `mocks/index.ts`